src/app/authentication/auth.ts

```diff
@@ -101,6 +101,23 @@ export function initAuth(options: IExplorerOptions, apiService: GraphService, ch
   handleAdminConsentResponse();
 }
 
+export function generateDefaultUserScopes(userScopes = AppComponent.Options.DefaultUserScopes) {
+  if (JSON.parse(localStorage.getItem('GRAPH_MODE')) === null) {
+    return userScopes;
+  }
+  const url = localStorage.getItem('GRAPH_URL');
+  const scopes = userScopes.split(' ');
+  let newScopes = ''
+  scopes.forEach(scope => {
+    if (scope === 'openid' || scope === 'profile') {
+      newScopes = newScopes + scope + ' '
+    } else {
+      newScopes = newScopes + url + scope + ' '
+    }
+  });
+  return newScopes
+}
```

Comment:
You can use a reduce function here

```
const reducedScope = scopes.reduce((newScopes, scope) => {
  if (scope === 'openid' || scope === profile) {
    return newScopes += scope + '';
  }
  return newScopes += url + scope + '';
}, '')
```

A reduce function doesn't mutate state therefore avoiding state related bugs.